spack/repo/packages/py-pyprecice/package.py

```diff
@@ -28,35 +30,25 @@ class PyPyprecice(PythonPackage):
     version("2.0.0.2", sha256="5f055d809d65ec2e81f4d001812a250f50418de59990b47d6bcb12b88da5f5d7")
     version("2.0.0.1", sha256="96eafdf421ec61ad6fcf0ab1d3cf210831a815272984c470b2aea57d4d0c9e0e")
 
-    # Older versions of the bindings checked versions via pip. This patch
-    # removes the pip dependency.
-    # See also https://github.com/spack/spack/pull/19558
-    patch("deactivate-version-check-via-pip.patch", when="@:2.1.1.1")
-
     depends_on("precice@develop", when="@develop")
+    depends_on("precice@2.2.1", when="@2.2.1.1:2.2.1.99")
     depends_on("precice@2.2.0", when="@2.2.0.1:2.2.0.99")
     depends_on("precice@2.1.1", when="@2.1.1.1:2.1.1.99")
     depends_on("precice@2.1.0", when="@2.1.0.1:2.1.0.99")
     depends_on("precice@2.0.2", when="@2.0.2.1:2.0.2.99")
     depends_on("precice@2.0.1", when="@2.0.1.1:2.0.1.99")
     depends_on("precice@2.0.0", when="@2.0.0.1:2.0.0.99")
 
-    depends_on("python@3:", type=("build", "run"))
+    depends_on("python@3:", type=("build", "link", "run"))
     depends_on("py-setuptools", type="build")
-    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "link", "run"))
     depends_on("py-mpi4py", type=("build", "run"))
-    depends_on("py-cython@0.29:", type=("build"))
-
-    phases = ['install_lib', 'build_ext', 'install']
-
-    def build_ext_args(self, spec, prefix):
-        return [
-            "--include-dirs=" + spec["precice"].headers.directories[0],
-            "--library-dirs=" + spec["precice"].libs.directories[0]
-        ]
-
-    def install(self, spec, prefix):
-        # Older versions of the bindings had a non-standard installation routine
-        # See also https://github.com/spack/spack/pull/19558#discussion_r513123239
-        if self.version <= Version("2.1.1.1"):
-            self.setup_py("install", "--prefix={0}".format(prefix))
+    depends_on("py-cython@0.29:", type="build")
+    depends_on("py-packaging", when="@:2.1", type="build")
+    depends_on("py-pip", when="@:2.1", type="build")
+
+    @when("@:2.1")
+    def patch(self):
+        filter_file(
+            "distutils.command.install", "setuptools.command.install", "setup.py"
+        )
```

Comment:
Does this change allow us to remove `spack/repo/packages/py-pyprecice/deactivate-version-check-via-pip.patch` without breaking older releases? If yes: Nice! If we are breaking older releases: No problem from my perspective, but we should document it in the Changelog.

Reply:
Yes, it does.  As mentioned in https://github.com/spack/spack/pull/25077 I have tested this with `py-pyprecice@2.0.0.1`, `py-pyprecice@2.1.1.2` and `pyprecice@2.2.1.1` which represents all installation procedures we had so far. 

- `2.0.0.1` is the initial installation process.
- In `2.0.0.2:2.1.1.2` there were some slight changes to the `setup.py`. 
- In `2.2.0.1:` versioneer was introduced which represents the current installation routine as far as I know.

Reply:
Great. Then this PR should be ready to be merged :rocket: